product_code/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala

```diff
@@ -65,7 +65,15 @@ class CourseServiceImpl(
             throw UC4Exception.OwnerMismatch
           }
 
-          val validationErrors = courseProposal.validate
+          // For syntax and regex checks, 5 seconds are more than enough
+          val validationErrors = try {
+            Await.result(courseProposal.validate, 5.seconds)
+          }
+          catch {
+            case _: TimeoutException => throw UC4Exception.ValidationTimeout
+            case e: Exception        => throw e
```

Comment:
This needs to be changed into a Internal Server Error
```suggestion
            case e: Exception        => throw UC4Exception.InternalServerError("Validation Error", e.getMessage)
```